airflow/utils/db.py

```diff
@@ -1004,15 +1010,17 @@ def reflect_tables(tables: list[Base | str] | None, session):
     """
     import sqlalchemy.schema
 
-    metadata = sqlalchemy.schema.MetaData(session.bind)
+    bind = session.bind
+    metadata = sqlalchemy.schema.MetaData()
+    metadata.bind = bind
 
     if tables is None:
-        metadata.reflect(resolve_fks=False)
+        metadata.reflect(bind=bind, resolve_fks=False)
     else:
         for tbl in tables:
             try:
                 table_name = tbl if isinstance(tbl, str) else tbl.__tablename__
-                metadata.reflect(only=[table_name], extend_existing=True, resolve_fks=False)
+                metadata.reflect(bind=bind, only=[table_name], extend_existing=True, resolve_fks=False)
```

Comment:
Is the `bind=bind` part still needed? You already set `metadata.bind` above.

Reply:
Have removed it. I remember having some issues around the metadata but let's see if the CI passes cause I can't reproduce locally

Reply:
Here's the error https://github.com/apache/airflow/actions/runs/5083011128/jobs/9133579355?pr=31486#step:5:10813, adding the bind back...

Reply:
What happened was that the metadata that was returned by the function was not bound to any database. The other binds are for the reflect methods. Despite the session being bound to metadata, they still want the bind to happen on its method

Reply:
Seeing this now...taking a look

Reply:
See https://github.com/apache/airflow/pull/31486/commits/8a08c05488e8583abc079fc238d518a00a87ca53, let me know what you think